src/SDKs/Azure.Base/data-plane/Azure.Base/Attributes/AzureSdkComponentAttribute.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Azure.Base.Attributes
+{
+    [AttributeUsage(AttributeTargets.Assembly, AllowMultiple = false)]
+    public class AzureSdkClientLibraryAttribute: Attribute
+    {
+        public string ComponentName { get; }
```

Comment:
It may be a worth having guidance for what kind of value to use for the component name or doing some light normalization to prevent different naming patterns.   Or, maybe that's over doing it and we should just leave it to PR feedback to keep consistency.    I'm kind of on the fence.    

If we're allowing any old pattern, should we consider just using the assembly name rather than a custom name, since people are unusually pretty good about consistent patterns for that?

Reply:
I feel that adding this attribute would happen so infrequently that it should be sanity checked during PRs